### DISCUSSION TITLE
Symbol File Issues in VS 2022 Build of OpenRadioss

### DISCUSSION THREAD

**Discussion by RiderCharles** [OpenRadioss/OpenRadioss#3089] [Q&A]
Hey, when I build OpenRadioss in VS 2022, neither engine_win64_db.exe nor starter_win64_db.exe generates the corresponding symbol files (.pdb). This means I can't debug the program even though breakpoints are set. How can I solve this problem? I look forward to your reply.

**Comment by OlivierWienholtz** [comment]
Hello, 
I do debug on VS2022 
The debug flags for Intel Fortran & C includes the symbols. There are no pdb files indeed.
when building in VS 2022 : I recommend build configurations 04 (Starter_x64-Debug) and 06 (Engine_x64_Debug)

Debugging in VS2022 is then possible.

**Reply by RiderCharles** [reply]
Hi, Olivier, I have set up build configurations 04 (Starter_x64-Debug) and 06 (Engine_x64_Debug), but the breakpoints are still not working. Do you have any other suggestions?
![Snipaste_2024-10-24_08-39-58](https://github.com/user-attachments/assets/776c0d9d-c169-46b7-b6ef-c5f3fb503c96)
![Snipaste_2024-10-24_08-40-19](https://github.com/user-attachments/assets/a172fdbf-a112-4afa-9859-92980a1b0f6e)

**Comment by Leo-Fakiya** [comment]
hello, I am facing the same issue. Hoping to hear back from you soon.Thanks.

**Reply by RiderCharles** [reply]
You could try compiling with VS2019 and refer to this link https://github.com/OpenRadioss/OpenRadioss/blob/main/doc/Visual_Studio_Debugger.md. 
That’s how I resolved the issue, and it might help you as well.

**Reply by Leo-Fakiya** [reply]
Thank you!!!!I seem to have solved the problem based on your suggestions, and it seems that VS2019 is the appropriate debug release.

**Comment by OlivierWienholtz** [comment]
Hello,
OpenRadioss relies on CMake to build binaries. It cannot produce .pdb files, but the Debug Version is built using Intel Compiler Flag
/debug:all
This permits debugging in Visual Studio 2022.
See Also : 
https://github.com/OpenRadioss/OpenRadioss/blob/main/doc/Visual_Studio_Debugger.md

**Reply by RiderCharles** [reply]
Hello, Olivier. Please forgive me for being occupied with other things over the past few days and not having had the chance to share updates with you. After switching to Visual Studio 2019, I was able to generate the .pdb file and successfully debug the program. Although I’m not entirely sure why it works on VS2019 and not on VS2022, I’m glad the issue is resolved.

**Comment by OlivierWienholtz** [comment]
Hello, 
Great that you where able to debug with VS2029.
